### DISCUSSION TITLE
Location of files and folders of 'Zero API'

### DISCUSSION THREAD

**Discussion by cloudcompute** [blitz-js/blitz#3172] [Q&A] [answered]
Pl tell me the names of the source files/folders for the code that implements the Zero API feature of the new toolkit.

**Comment by cloudcompute** [comment, marked as answer]
I had a look a the "main" branch. It looks like the work has not started yet for the Zero API (RPC) feature for which Blitz got popularity.

The proposal had come 2 months back and almost nothing is there in the "main" branch. Initially the commitment was made for the month of February and then postponed to March. :(

**Reply by beerose** [reply]
Hey! Yeah, we're sorry about that. The work _is_ in progress, but it's a lot of work, so things are taking longer than expected. Thanks for your patience!

**Reply by prisis** [reply]
Its still a open source project (not a paid one!), so it can happen that something is not delivered at promised. If you want to have something "faster" so please help to develop it.

**Comment by cloudcompute** [comment]
@prisis

I am of the viewpoint that if it is an open-source package owned by one or two individuals and they suddenly decided not to support it, we can understand.

But if an open source package has

- 360 contributors.. even their photographs are there
- 10K+ stars
- 2390 commits
- 211 releases
- first relased was made on Apr 2020, it has been almost two years since now
- used in production (see Blitz website) I can recall @flybaer mentioning somewhere in the discussion that "hundreds" of developers have been using Blitz in production. Imagine the plight of those projects after you guys proposed the new kit.

My observation.. Blitz was aiming for sponsors right from the beginning. A lot of hype was created around this project since it began. Because of this, you were able to make a place in the JavaScript world. Bloggers started writing that Blitz is an upcoming full-stack framework, compared it with Redwood and even Next JS. After reading all this, naturally, we will start expecting something big from Blitz. But unfortunately, Blitz is back to from where it started.

If more delays in the new Blitz release are going to be there, I think Blitz will loose its place in the fast-changing JavaScript ecosystem soon. In my previous post, what I wanted to say was... Blitz is going through a sensitive time, therefore, try to keep the promises.

As far as my contributions are involved, I might help in the architect and design of the new Blitz.

**Reply by prisis** [reply]
@ramandhingra A project with millions of downloads per month can decided to close or stop to develop it, see faker.js or color.js, all of this stuff is still open source and the people spending FREE time on it to get the stuff out. 

With faith you can't buy anything... so you need to set other priorities first.

If you can't wait for the feature like the 0 api, you could build it yourself with https://trpc.io/docs/nextjs, prisma and use the nextjs auth package and so on.

**Comment by cloudcompute** [comment]
@prisis 

Katt was invited in the "new toolkit" discussion. From there, I got to know that there exists an RPC-enabled package named trpc. I am going through it and most probably I'll use it in the forthcoming project. Looks like it is stable enough v9.18.
